### DISCUSSION TITLE
Difficulty Creating Buckets in Apache Ozone via REST API

### DISCUSSION THREAD

**Discussion by FaheemKhan0817** [apache/ozone#6619] [FAQ] [answered]
Dear Apache Ozone Community,

I hope this message finds you well. I am reaching out to share a challenge I am facing while attempting to create buckets in Apache Ozone using the REST API.

Specifically, I have been trying to create a new bucket within a volume using Python and the requests library to interact with the Apache Ozone REST API. However, despite following the documentation and guidelines available, I have encountered difficulties in successfully creating buckets.
#####---> replace inplace correct url
Here is the Python code snippet I have been using:

python
Copy code
import requests

# Apache Ozone Recon API base URL
OZONE_API_URL = "http://########:9888/api/v1"

# Volume name
volume_name = "vol1"

# Bucket name to be created
new_bucket_name = "new_bucket"

# Endpoint for creating a bucket in a volume
endpoint = f"{OZONE_API_URL}/volume/{volume_name}/bucket/{new_bucket_name}"

# Send a PUT request to create the bucket
response = requests.put(endpoint)

# Check if the bucket was created successfully
if response.status_code == 200:
    print(f"Bucket '{new_bucket_name}' created successfully in volume '{volume_name}'.")
else:
    print(f"Failed to create bucket. Status code: {response.status_code}")
    print(response.text)  # Print the response content for further inspection if needed
Despite multiple attempts and adjustments, I have been consistently receiving error responses, indicating a failure to create the bucket. I have double-checked the endpoint URL, volume name, and bucket name, but the issue persists.

I would greatly appreciate any insights, guidance, or assistance that the community can provide to help resolve this issue. Additionally, if there are any specific configurations or permissions that need to be set up in Apache Ozone to enable bucket creation via the API, I would be grateful for any instructions or recommendations in this regard.

Thank you very much for your attention and support. I look forward to hearing from you and resolving this issue together.

Best regards,
Faheem Khan

**Comment by smengcl** [comment, marked as answer]
Hi Faheem, it looks like you are trying to create a bucket via Recon. This is not supposed to work because Ozone Recon (and its API) only provides Ozone cluster status but not file system read/write operations. More details on Recon APIs: https://ozone.apache.org/docs/current/interface/reconapi.html

You should use Ozone S3 Gateway for this purpose instead, please check its documentation here: https://ozone.apache.org/docs/current/interface/s3.html

**Comment by azro352** [comment]
@FaheemKhan0817 

Use 3 backticks `` ` ``  around your lines of codes so it's properly indented

like this

````
```python
code
code
code
```
```` 

edit your initial post to try it , It should looks like 

```python
import requests

# Apache Ozone Recon API base URL
OZONE_API_URL = "http://########:9888/api/v1"

# Volume name
volume_name = "vol1"

# Bucket name to be created
new_bucket_name = "new_bucket"
```

**Comment by spacemonkd** [comment]
Closing this as this has been inactive for a long time. @FaheemKhan0817, if this issue is not resolved for you, you can reopen the discussion.
Thanks
